spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -124,23 +128,47 @@ public void afterUnbindConsumer(ConsumerDestination destination) {
 		}
 	}
 
-	private Optional<Topic> ensureTopicExists(String topicName, boolean autoCreate) {
+	Topic ensureTopicExists(String topicName, boolean autoCreate) {
```

Comment:
I like the recursion, clever :+1:

Minor refactor suggestion: I would suggest first checking if `topic != null` and then returning early instead of doing the `topic == null` check. To me it looks a little bit simpler.

```
Topic topic = this.pubSubAdmin.getTopic(topicName);

if (topic != null) {
  return topic;
}

if (autoCreate) {
	try {
		return this.pubSubAdmin.createTopic(topicName);
	}
	catch (AlreadyExistsException alreadyExistsException) {
		return ensureTopicExists(topicName, false);
	}
}
else {
	throw new ProvisioningException("Non-existing '" + topicName + "' topic.");
}

```